kip-0017.md

```diff
@@ -8,105 +8,422 @@ Category: Interface
+This method returns the Kadena, fungible account names associated with a specific Wallet Connect account (a prefixed public key).
 
-## SLIP-0044
-While not required in the namespace proposal process, a network's SLIP-0044 value is used throughout the wallet connect demo code.
-Kadena's coin type is 626 as documented in https://github.com/satoshilabs/slips/blob/master/slip-0044.md
+These Wallet Connect accounts are retrieved from the `accounts` property of a [Paring response](#pairing-with-walletconnect).
 
-# Example Proposal and Response
+#### Example kadena_getAccounts_v1 Request and Response
 
-```json
-# Proposal namespace request
+```jsonc
+// kadena_getAccounts_v1 Method Request
 {
-  "kadena": {
-    "chains": ["kadena:mainnet01", "kadena:testnet04", "kadena:development"],
-    "methods": ["kadena_quicksign_v1"],
-    "events": []
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "kadena_getAccounts_v1",
+  "params": {
+    "accounts": [
+      {
+        "account": "kadena:mainnet01:38298612cc2d5e841a232bd08413aa5304f9ef3251575ee182345abc3807dd89",
+        "contracts": ["coin"] // optional, when omitted, the wallet returns all known fungible accounts
+      }
+    ]
   }
 }
+````
+
+```jsonc
+// kadena_getAccounts_v1 Method Response
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "result": [
+    {
+      "accounts": [
+        {
+          "account": "kadena:mainnet01:38298612cc2d5e841a232bd08413aa5304f9ef3251575ee182345abc3807dd89",
+          "publicKey": "38298612cc2d5e841a232bd08413aa5304f9ef3251575ee182345abc3807dd89",
+          "kadenaAccounts": [
+            {
+              "name": "w:aoriestnaoirsetnaorisetn",
+              "contract": "coin",
+              "chains": ["0", "1"]
+            },
+            {
+              "name": "k:38298612cc2d5e841a232bd08413aa5304f9ef3251575ee182345abc3807dd89",
+              "contract": "coin",
+              "chains": [
+                "0",
+                "1",
+                "2",
+                "3",
+                "4",
+                "5",
+                "6",
+                "7",
+                "8",
+                "9",
+                "10",
+                "11",
+                "12",
+                "13",
+                "14",
+                "15",
+                "16",
+                "17",
+                "18",
+                "19"
+              ]
+            }
+          ]
+        }
+      ]
+    }
+  ]
+}
+```
+
+#### kadena_getAccounts_v1 Request
+
+This method needs to be called with a `params` object with the following properties:
+
+1. `params`: `Object`\
+    1.1. `accounts`: `Array<Object>`\
+    __ 1.1.1 `account`: `string` - the public key (a Wallet Connect [Account](#wallet-connect-accounts)) for which the Kadena account name is requested.\
+    __ 1.1.2 `contracts`: `Array<string>` (optional) - array of fungible contracts for which the Kadena account name is requested. Returns all known Kadena accounts if omitted.
```

Comment:
I used the word "fungible" in a couple places, but I'm not sure if this should just be omitted since there might be other use cases where you want to retrieve the account name but not its not a fungible contract account name.

Reply:
The account name is coming from the ledger in a fungible contract, so there's always a fungible in play here.